### ISSUE TITLE
Separate Realized P&L from Unrealized P&L: Add current holdings valuation from Yahoo Finance

### ISSUE THREAD

**Issue by zheharry** [zheharry/my-portfolio#49]
## Problem

Currently, the "Realized P&L" calculation includes ALL transactions, but it should **only calculate sold quantities**. We need to separate:

1. **Realized P&L**: Only from SOLD positions (actual cash gains/losses)
2. **Unrealized P&L**: Current holdings valued at market prices (paper gains/losses)

## Current Issue ❌

**Realized P&L** currently calculates:
```sql
SUM(CASE WHEN transaction_type = '賣出' THEN net_amount ELSE 0 END) as total_sales,
SUM(CASE WHEN transaction_type = '買進' THEN ABS(net_amount) ELSE 0 END) as total_purchases
realized_gain_loss = total_sales - total_purchases  -- ❌ Includes unsold holdings
```

This mixes realized (sold) and unrealized (held) positions, providing inaccurate financial analysis.

### 1. **Fix Realized P&L Calculation**
- **Only calculate gains/losses from SOLD quantities**
- Use matched buy/sell pairs to determine actual realized gains
- Exclude current holdings from realized P&L

### 2. **Add Unrealized P&L Card**
- **New card-body next to Realized P&L**
- Calculate current market value of held positions
- Show potential gains/losses on unsold holdings

### 3. **Yahoo Finance Integration**
- Fetch current/last trade day prices for held symbols
- No need for real-time data - end-of-day prices are sufficient
- Handle multiple currencies (USD for US stocks, NTD for Taiwan stocks)

### Database Analysis Required
```sql
-- Get current holdings (bought - sold quantities)
SELECT 
    symbol,
    SUM(CASE WHEN transaction_type IN ('買進', 'BUY') THEN quantity ELSE 0 END) as bought_qty,
    SUM(CASE WHEN transaction_type IN ('賣出', 'SELL') THEN ABS(quantity) ELSE 0 END) as sold_qty,
    SUM(CASE WHEN transaction_type IN ('買進', 'BUY') THEN quantity 
             WHEN transaction_type IN ('賣出', 'SELL') THEN -ABS(quantity) ELSE 0 END) as current_holding,
    AVG(CASE WHEN transaction_type IN ('買進', 'BUY') THEN price ELSE NULL END) as avg_cost
FROM transactions 
WHERE symbol IS NOT NULL 
GROUP BY symbol
HAVING current_holding > 0
```

### Yahoo Finance API Integration
```python
import yfinance as yf

def get_current_prices(symbols):
    """Fetch current prices for symbols from Yahoo Finance"""
    prices = {}
    for symbol in symbols:
        try:
            ticker = yf.Ticker(symbol)
            hist = ticker.history(period="1d")
            if not hist.empty:
                prices[symbol] = hist['Close'].iloc[-1]
        except:
            prices[symbol] = None
    return prices
```

### New API Endpoint
```python
@app.route('/api/unrealized-pnl')
def api_unrealized_pnl():
    """Calculate unrealized P&L for current holdings"""
    holdings = get_current_holdings()
    symbols = [h['symbol'] for h in holdings]
    current_prices = get_current_prices(symbols)
    
    unrealized_pnl = 0
    for holding in holdings:
        symbol = holding['symbol']
        quantity = holding['current_holding']
        avg_cost = holding['avg_cost']
        current_price = current_prices.get(symbol, avg_cost)
        
        market_value = quantity * current_price
        cost_basis = quantity * avg_cost
        unrealized_pnl += (market_value - cost_basis)
    
    return jsonify({
        'unrealized_pnl': unrealized_pnl,
        'total_market_value': sum(h['market_value'] for h in holdings),
        'total_cost_basis': sum(h['cost_basis'] for h in holdings)
    })
```

### Updated UI Cards
```html
<- **Future-proof filtering** as new brokers are added Existing Realized P&L Card -->
<div class="col-md-3">
    <div class="card summary-card">
        <div class="card-body text-center">
            <h5 class="card-title">已實現損益 (Realized P&L)</h5>
            <h6 class="text-muted">僅已賣出部位 (Sold positions only)</h6>
            <h3 id="realizedPL">bash</h3>
        </div>
    </div>
</div>

<- **Future-proof filtering** as new brokers are added NEW Unrealized P&L Card -->
<div class="col-md-3">
    <div class="card summary-card">
        <div class="card-body text-center">
            <h5 class="card-title">未實現損益 (Unrealized P&L)</h5>
            <h6 class="text-muted">目前持股 (Current holdings)</h6>
            <h3 id="unrealizedPL">bash</h3>
            <small class="text-muted">基於最新收盤價 (Based on latest closing prices)</small>
        </div>
    </div>
</div>
```

### Financial Accuracy
- **True realized P&L**: Only actual gains/losses from completed trades
- **Market-based unrealized P&L**: Current value of held positions
- **Better investment analysis**: Separate paper vs. actual returns

### User Experience  
- **Clear distinction** between sold and held positions
- **Current portfolio value** based on market prices
- **Comprehensive view** of both realized and potential gains/losses

### Phase 1: High Priority
- [ ] Fix realized P&L calculation (only sold quantities)
- [ ] Add basic unrealized P&L calculation using average cost
- [ ] Create new unrealized P&L card in UI

### Phase 2: Medium Priority
- [ ] Integrate Yahoo Finance API for current prices
- [ ] Handle multiple currencies (USD/NTD)
- [ ] Add error handling for API failures

### Phase 3: Low Priority
- [ ] Add caching for price data (update daily)
- [ ] Add portfolio performance metrics
- [ ] Add historical unrealized P&L tracking

## Dependencies

- **yfinance** package: `pip install yfinance`
- **API rate limiting**: Yahoo Finance has usage limits
- **Symbol mapping**: Handle symbol differences between brokers

## Expected Impact

- **More accurate financial reporting**
- **Better investment decision making** 
- **Professional-grade portfolio analysis**
- **Clear separation of realized vs. unrealized performance**

This enhancement will provide users with accurate, market-based portfolio valuation alongside their actual trading performance.

**Comment by zheharry** [comment]
## 🔍 **Current Code Analysis - Confirmed Issue**

I've analyzed the current implementation and **confirmed the problem**:

### ❌ **Current Incorrect Calculation** (Lines 507-520 in app.py)

```python
# Current logic - WRONG approach
if transaction_type == '賣出':
    total_sales_ntd += net_amount_ntd
elif transaction_type == '買進':
    total_purchases_ntd += abs(net_amount_ntd)

# Then calculates: ❌ INCORRECT
realized_gain_loss_ntd = total_sales_ntd - total_purchases_ntd
```

**Problem**: This calculation includes **ALL purchases** (including unsold holdings) vs **only sold amounts**, which gives a misleading "realized" P&L.

### 📊 **Real-World Example of the Problem**

**Scenario**: User bought 1000 shares of 台積電 at 00, sold 300 shares at 20, still holds 700 shares

**Current Calculation**:
- Total Sales: 300 × 20 = 6,000
- Total Purchases: 1000 × 00 = 00,000  
- "Realized" P&L: 6,000 - 00,000 = **-4,000** ❌

**Correct Calculation Should Be**:
- Actually Realized P&L: (300 × 20) - (300 × 00) = **+,000** ✅
- Unrealized P&L: (700 × current_price) - (700 × 00) = **[Market dependent]**

### 🎯 **Database Evidence**

From actual database query:
```sql
-- Current holdings example
台積電|4805 bought|3000 sold  --> 1805 shares still held
```

The current calculation incorrectly includes the cost of those 1805 unsold shares in "realized" P&L.

### ✅ **Required Fix Summary**

1. **Realized P&L**: Only calculate on matched buy/sell pairs for SOLD quantities
2. **Unrealized P&L**: Calculate current market value of remaining holdings using Yahoo Finance
3. **Separate Cards**: Display both metrics clearly in the UI

This will provide users with accurate financial reporting instead of the current misleading figures.
